caddyhttp/browse/browse.go

```diff
@@ -333,9 +333,14 @@ func (b Browse) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
 
 	// Browsing navigation gets messed up if browsing a directory
 	// that doesn't end in "/" (which it should, anyway)
-	if !strings.HasSuffix(r.URL.Path, "/") {
-		staticfiles.RedirectToDir(w, r)
-		return 0, nil
+	u := r.Context().Value(httpserver.OriginalURLCtxKey).(url.URL)
```

Comment:
Yeah, good catch. This is tricky. The reason I chose to get the path of the original URL is because we end up causing a redirect with it; the redirect is external, but the rewrite is internal.

Maybe... should we check for a trailing slash using `r.URL.Path` and then issue the direct by modifying the original path stored in the context?